sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableConfigTranslator.java

```diff
@@ -149,7 +154,45 @@ private static BigtableDataSettings.Builder buildBigtableDataSettings(
     configureChannelPool(dataBuilder.stubSettings(), config);
     configureHeaderProvider(dataBuilder.stubSettings(), pipelineOptions);
 
-    return dataBuilder;
+    // Provide a way to override any BigtableDataSettings
+    String overrideClassName =
+        ExperimentalOptions.getExperimentValue(pipelineOptions, BIGTABLE_SETTINGS_OVERRIDE);
+
+    return configureSettingsOverride(overrideClassName, dataBuilder, pipelineOptions);
+  }
+
+  @SuppressWarnings("unchecked")
+  private static BigtableDataSettings.Builder configureSettingsOverride(
+      @Nullable String override,
+      BigtableDataSettings.Builder dataBuilder,
+      PipelineOptions pipelineOptions) {
+    if (override == null) {
+      return dataBuilder;
+    }
+    Object object;
```

Comment:
I could have sworn there was a utility method in beam somewhere that did this, but I can't find it now :(

Reply:
Ah ha: not blocking but you might want to look into using this instead to fit into the normal beam pattern of doing this:  https://github.com/apache/beam/blob/b6301b5205800ce3604a751964211d6061b09f02/sdks/java/core/src/main/java/org/apache/beam/sdk/util/InstanceBuilder.java